include/drivers/can.h

```diff
@@ -766,8 +799,21 @@ static inline int can_set_bitrate(const struct device *dev,
 #ifdef CONFIG_CAN_FD_MODE
 	struct can_timing timing_data;
 #endif
+	uint32_t max_bitrate;
 	int ret;
 
+	ret = can_get_max_bitrate(dev, &max_bitrate);
+	if (ret == -ENOSYS) {
+		/* Maximum bitrate unknown */
+		max_bitrate = 0;
+	} else if (ret < 0) {
+		return ret;
+	}
+
+	if ((max_bitrate > 0) && (bitrate > max_bitrate)) {
+		return -ENOTSUP;
+	}
+
```

Comment:
This function is getting pretty long for a `static inline`. See also recommendations in [Linux Kernel coding style guide](https://www.kernel.org/doc/html/latest/process/coding-style.html#the-inline-disease). Should we move it to `can_common.c`?

Reply:
Good idea. I have moved it to `can_common.c` in a separate commit to make reviewing of the changes easier.